core/domain/prod_validation_jobs_one_off_test.py

```diff
@@ -4967,6 +4969,1325 @@ def test_model_with_assymetric_content(self):
+            u'[u\'fully-validated SkillModel\', 4]']
+        run_job_and_check_output(self, expected_output, sort=True)
+
+    def test_missing_snapshot_content_model_failure(self):
+        skill_models.SkillSnapshotContentModel.get_by_id(
+            '0-1').delete()
+        expected_output = [
+            (
+                u'[u\'failed validation check for snapshot_content_ids '
+                'field check of SkillModel\', '
+                '[u"Entity id 0: based on field snapshot_content_ids having '
+                'value 0-1, expect model SkillSnapshotContentModel '
+                'with id 0-1 but it doesn\'t exist"]]'),
+            u'[u\'fully-validated SkillModel\', 4]']
+        run_job_and_check_output(self, expected_output, sort=True)
+
+
+class SkillSnapshotMetadataModelValidatorTests(
+        test_utils.GenericTestBase):
+
+    def setUp(self):
+        super(SkillSnapshotMetadataModelValidatorTests, self).setUp()
+
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.signup(USER_EMAIL, USER_NAME)
+
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.user_id = self.get_user_id_from_email(USER_EMAIL)
+
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.set_admins([self.ADMIN_USERNAME])
+
+        language_codes = ['ar', 'en', 'en']
+        skills = [skill_domain.Skill.create_default_skill(
+            '%s' % i,
+            description='description %d' % i
+        ) for i in xrange(3)]
+
+        skill_contents = skill_domain.SkillContents(
+            state_domain.SubtitledHtml(
+                '1', '<p>Explanation</p>'), [
+                    state_domain.SubtitledHtml('2', '<p>Example 1</p>')],
+            {'1': {}, '2': {}}, state_domain.WrittenTranslations.from_dict(
+                {'translations_mapping': {'1': {}, '2': {}}}))
+        misconception_dict = {
+            'id': 0, 'name': 'name', 'notes': '<p>notes</p>',
+            'feedback': '<p>default_feedback</p>'}
+
+        for index, skill in enumerate(skills):
+            skill.language_code = language_codes[index]
+            skill.skill_contents = skill_contents
+            skill.add_misconception(misconception_dict)
+            if index == 0:
+                skill_services.save_new_skill(self.user_id, skill)
+            else:
+                skill_services.save_new_skill(self.owner_id, skill)
+
+        self.model_instance_0 = (
+            skill_models.SkillSnapshotMetadataModel.get_by_id(
+                '0-1'))
+        self.model_instance_1 = (
+            skill_models.SkillSnapshotMetadataModel.get_by_id(
+                '1-1'))
+        self.model_instance_2 = (
+            skill_models.SkillSnapshotMetadataModel.get_by_id(
+                '2-1'))
+
+        self.job_class = prod_validation_jobs_one_off.SkillSnapshotMetadataModelAuditOneOffJob # pylint: disable=line-too-long
```

Comment:
See @brianrodri's comments above -- you can break on the period. Ditto elsewhere.

Reply:
Done.